pcf/particle/aws/iam/iam_policy.py

```diff
@@ -0,0 +1,169 @@
+
+    def _set_unique_keys(self):
+        """
+        Logic that sets keys from state definition that are used to uniquely identify IAM Policies
+        """
+        self.unique_keys = IAMPolicy.UNIQUE_KEYS
+
+    def get_status(self):
+        """
+        Determines if the IAM Policy exists
+        Returns:
+             status (dict)
+        """
+
+        try:
+            policy_status = self.client.list_policies(Scope="All")
+        except ClientError as e:
+            if e.response['Error']['Code'] == 'PolicyNotFoundException':
+                logger.warning("Policy {} was not found. Defaulting state for {} to terminated".format(self.policy_name, self.policy_name))
+                return {"status": "missing"}
+            else:
+                raise e
+
+        policy = [x for x in policy_status.get('Policies') if x.get("PolicyName") == self.policy_name]
+
+        if policy:
+            if policy[0].get('Arn'):
+                self.policy_arn = policy[0].get('Arn')
+            return {"status":"active"}
+        else:
+            return {"status": "missing"} 
+
+    def _terminate(self):
+        """
+        Deletes the IAM Policy
+        Returns:
+             response of boto3 delete_policy
+        """
+        policy_versions = self.client.list_policy_versions(
+            PolicyArn=self.policy_arn,
+        )
+
+        # All policy versions must be deleted before default policy can be deleted. 
+        if policy_versions:
+            for v in policy_versions.get('Versions'):
+                if(not v.get('IsDefaultVersion')):
+                    self.client.delete_policy_version(PolicyArn=self.policy_arn, VersionId=v.get('VersionId'))
+
+        return self.client.delete_policy(PolicyArn=self.policy_arn)
+
+    def _start(self):
+        """
+        Creates the IAM Policy
+        Returns:
+             response of boto3 create_policy
+        """
+        create_definition = pcf_util.param_filter(self.get_desired_state_definition(), IAMPolicy.START_PARAMS_FILTER)
+
+        return self.client.create_policy(**create_definition)
+
+    def _stop(self):
+        """
+        IAM Policy does not have a stopped state so it calls terminate.
+        """
+        return self.terminate()
+
+    def sync_state(self):
+        """
+        Calls get status and then sets the current state.
+        """
+        full_status = self.get_status()
+
+        if full_status:
+            status = full_status.get("status", "missing").lower()
+            self.state = IAMPolicy.state_lookup.get(status)
+
+        if full_status.get('status') != 'missing':
+            policy = self.client.get_policy(
+                PolicyArn = self.policy_arn
+            )
+
+            policy_version = self.client.get_policy_version(
+                PolicyArn=self.policy_arn,
+                VersionId= policy['Policy']['DefaultVersionId']
+            )
+
+            a = str(policy_version.get('PolicyVersion').get('Document'))
```

Comment:
instead of str() and replace maybe try json.dumps() instead. this is the weird error in is state def equivalent right?

Reply:
oh yeah good catch. Same issue as iam_role forgot to change it for policy.